lib/temp.rb

```diff
@@ -0,0 +1,34 @@
+
+def temp_exists?(name)
+  File.exists?(temp_join(name))
+end
+
+def temp_join(name)
+  File.join(Dir.pwd, "tmp", name)
+end
+
+def temp_memoize(key)
+  if temp_exists?(key)
+    temp_read(key)
+  else
+    value = yield
+
+    temp_write(key, value)
+
+    value
+  end
+end
+
+def temp_read(name)
+  lines =
+    File.readlines(temp_join(name))
+    .map { |line| line.strip }
+
+  lines.length > 1 ? lines : lines.first
```

Comment:
I'd call it generally an anti-pattern to return two different types from a function. Here we're returning either an array of strings or a string. Did you mean to join the lines with "\n"?
```ruby
def temp_read(name)
  File.readlines(temp_join(name)).map(&:strip).join("\n")
end
```

Reply:
[Reference](https://www.brianstorti.com/understanding-ruby-idiom-map-with-symbol/) for the `&:` shorthand, in case anyone hasn't seen it before.

Reply:
There's a scenario where we need to memoize an array of strings and memoize an individual string, at least when I wrote the first version. I'm not sure if that's necessary anymore.

Reply:
Inlined all the functions that aren't used outside of `temp.rb` and `temp_memoize` now always returns an array.